src/main/java/it/gov/pagopa/reward/notification/service/exports/ForceOrganizationExportServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package it.gov.pagopa.reward.notification.service.exports;
+
+import it.gov.pagopa.reward.notification.model.RewardOrganizationExport;
+import it.gov.pagopa.reward.notification.repository.RewardOrganizationExportsRepository;
+import it.gov.pagopa.reward.notification.service.csv.out.ExportRewardNotificationCsvService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@Slf4j
+public class ForceOrganizationExportServiceImpl implements ForceOrganizationExportService {
+
+    private final RewardOrganizationExportsRepository rewardOrganizationExportsRepository;
+    private final ExportRewardNotificationCsvService exportRewardNotificationCsvService;
+
+
+    public ForceOrganizationExportServiceImpl(RewardOrganizationExportsRepository rewardOrganizationExportsRepository, ExportRewardNotificationCsvService exportRewardNotificationCsvService) {
+        this.rewardOrganizationExportsRepository = rewardOrganizationExportsRepository;
+        this.exportRewardNotificationCsvService = exportRewardNotificationCsvService;
+    }
+
+    @Override
+    public Flux<List<RewardOrganizationExport>> execute(LocalDate notificationDateToSearch) {
+        LocalDate now = LocalDate.now();
+
+        return rewardOrganizationExportsRepository.findByExportDate(now)
+                .flatMap(this::cleanTodayExport)
+                .thenMany(exportRewardNotificationCsvService.execute(notificationDateToSearch));
```

Comment:
here I have a doubt...
we should proceed with the exportRewardNotificationCsvService after all the "findByExportDate" results have been processed...
the "thenMany" probably is executed at each "findByExportDate" result istead of once at its end (I think this because I see you are forced to use thenMany and not simply then, thus I thing it's expecting to have a Mono a return type)
```suggestion
                .collectList()
                .then(exportRewardNotificationCsvService.execute(notificationDateToSearch));
```

Reply:
We could write a test to let the findByExportDate to return more than 1 object, verifying that the exportRewardNotificationCsvService has been called exactly once